File: monitoring/src/main/java/com/example/monitoring/user/adapter/port/in/response/UserResponse.java
package com.example.monitoring.user.adapter.port.in.response;

import com.example.monitoring.user.application.port.in.response.UserResponseDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {

    private Long id;
    private String name;
    private int age;

    public static UserResponse from(UserResponseDto userResponseDto) {
        return UserResponse.builder()
                .id(userResponseDto.getId())
                .name(userResponseDto.getName())
                .age(userResponseDto.getAge())
                .build();
    }
}

File: kotlin-spring-boot/src/main/kotlin/com/hello/kotlin/user/presentation/response/UserResponse.kt
package com.hello.kotlin.user.presentation.response

import com.hello.kotlin.user.application.response.UserResponseDto

class UserResponse(
    val id: Long,
    val name: String,
    val age: Int,
    val address: String,
) {
    companion object {
        fun from(responseDto: UserResponseDto) : UserResponse =
            UserResponse(responseDto.id,
                responseDto.name,
                responseDto.age,
                responseDto.address
            )
    }
}
